File: dividir_conquistar/dividir_conquistar.exs
defmodule DividirConquistar do
  def soma([]), do: 0
  def soma([head | nil]), do: head
  def soma([head | tail]), do: head + soma(tail)

  def conta([]), do: 0
  def conta([_ | nil]), do: 1
  def conta([_ | tail]), do: 1 + conta(tail)

  def maior([]), do: 0
  def maior([head | nil]), do: head
  def maior([head | tail]), do: if head > maior(tail), do: head, else: maior(tail)
end

IO.puts(DividirConquistar.soma([1, 4, 6, 2, 5, 9]))
IO.puts(DividirConquistar.conta([1, 4, 6, 2, 5, 9]))
IO.puts(DividirConquistar.maior([1, 4, 6, 7, 5, 3]))

File: dividir_conquistar/dividir_conquistar.py
def soma(x):
    if x == []:
        return 0
    else:
        return x[0] + soma(x[1:])

print(soma([1, 2, 3, 4]))
print(soma([]))
print(soma([1]))


def conta(x):
    if x == []:
        return 0
    return 1 + conta(x[1:])

print(conta([1, 5, 2, 43, 123, 45]))
print(conta([1]))
print(conta([]))

def maior(x):
    if len(x) == 1:
        return x[0]
    return x[0] if x[0] > maior(x[1:]) else maior(x[1:])

print(maior([3, 7, 5, 8]))
